### PR TITLE
Add vectorized operation tests

### PR DESCRIPTION
**Pull request by blegat**
Commented tests in JuMP v0.18 have been uncommented. One of them mentionned that was failing because of #912 but it now works. One of them was failing because of #1275 but a fix is contained in this PR.
A last test is failing because of https://github.com/JuliaOpt/JuMP.jl/issues/1276 but I defer the fix for a later PR.

Closes https://github.com/JuliaOpt/JuMP.jl/issues/912
Closes https://github.com/JuliaOpt/JuMP.jl/issues/1275

### CODE REVIEW

File: test/operator.jl [hunk 1]
```diff
@@ -247,4 +247,281 @@ using Compat.Test
...
+            @test vec_eq(X.'*A, X'*A)
+            @test vec_eq(A.'*X, A'*X)
+            @test vec_eq(X*A, X*B)
+            @test vec_eq(Y'*A, Y.'*A)
+            @test vec_eq(A*Y', A*Y.')
+            @test vec_eq(Z'*A, Z.'*A)
+            @test vec_eq(Xd'*Y, Xd.'*Y)
+            @test vec_eq(Y'*Xd, Y.'*Xd)
+            @test vec_eq(Xd'*Xd, Xd.'*Xd)
+            # @test_broken vec_eq(A*X, B*X)
```
Comment (mlubin) [inline test/operator.jl]: Why is this code commented?

File: test/operator.jl [hunk 2]
```diff
@@ -247,4 +247,281 @@ using Compat.Test
...
+            m = Model()
+            @variable(m, x[1:2,1:2])
+            A = [1 2;
+                 3 4]
+            B = sparse(A)
+            y = SparseMatrixCSC(2, 2, copy(B.colptr), copy(B.rowval), vec(x))
+            @test vec_eq(A.+x, [1+x[1,1]  2+x[1,2];
+                                3+x[2,1]  4+x[2,2]])
+            @test vec_eq(A.+x, B.+x)
+            # @test vec_eq(A.+x, A.+y)
```
Comment (mlubin) [inline test/operator.jl]: Why is this commented?

**Comment by mlubin** [comment]
Nice boost in coverage for `src/operators.jl`!

File: test/operator.jl (hunk 1 above)
Comment (blegat) [reply test/operator.jl]: It was commented out in `old/operators.jl`, I have uncommented them and checked what were the issue, see the description of this PR for details :)

**Pull request merged**